src/endpoints/authentication/impl.cpp

```diff
@@ -20,120 +21,484 @@
+					nlohmann::json res_item{hit_token_endpoint(encode_body(args))};
+
+					// Assume passed, get oidc token
+					const std::string jwt_token{res_item.at("id_token").get_ref<const std::string&>()};
+
+					// Get OIDC token && feed to JWT parser
+					// TODO: Handle case where we throw!!!
+					auto decoded_token{
+						jwt::decode<jwt::traits::nlohmann_json>(res_item.at("id_token").get_ref<const std::string&>())};
+
+					// Get irods username
+					const std::string irods_name{
+						decoded_token.get_payload_json().at("irods_username").get<const std::string>()};
+
+					// Issue token?
+					static const auto seconds =
+						irods::http::globals::configuration()
+							.at(nlohmann::json::json_pointer{"/http_server/authentication/basic/timeout_in_seconds"})
+							.get<int>();
+
+					auto bearer_token = irods::process_stash::insert(authenticated_client_info{
+						.auth_scheme = authorization_scheme::basic,
+						.username = std::move(irods_name),
+						.expires_at = std::chrono::steady_clock::now() + std::chrono::seconds{seconds}});
+
+					response_type res_rep{status_type::ok, _req.version()};
+					res_rep.set(field_type::server, BOOST_BEAST_VERSION_STRING);
+					res_rep.set(field_type::content_type, "text/plain");
+					res_rep.keep_alive(_req.keep_alive());
+					res_rep.body() = std::move(bearer_token);
+					res_rep.prepare_payload();
+
+					return _sess_ptr->send(std::move(res_rep));
+				});
+			}
+		}
+		// Handle posts
+		else if (_req.method() == boost::beast::http::verb::post) {
+			irods::http::globals::background_task([fn = __func__, _sess_ptr, _req = std::move(_req)] {
+				// Right, we're kinda being a proxy, so how about proxy-authorization?
+				const auto& hdrs{_req.base()};
+				const auto iter{hdrs.find("authorization")};
+
+				// Failed to find auth header, use OIDC auth_code flow instead
+				if (iter == std::end(hdrs)) {
+					return _sess_ptr->send(fail(status_type::bad_request));
+				}
+
+				log::debug("{}: Authorization value: [{}]", fn, iter->value());
+
+				//
+				// TODO Here is where we determine what form of authentication to perform (e.g. Basic or OIDC).
+
+				//
+				// BLAH BLAH BLAH, assume we have the user & pass...
+				// Prob via Proxy-Authorization????
+				//
+
+				const auto pos = iter->value().find("Basic ");
+				if (std::string_view::npos == pos) {
+					// TEMPORARY PLS MAKE BETTER LATER :)
+
+					const auto alt_method{iter->value().find("iRODS ")};
+					if (std::string_view::npos == alt_method) {
+						return _sess_ptr->send(fail(status_type::bad_request));
+					}
+
+					// BEGIN BASE64 HEADER DECODE
+
+					constexpr auto basic_auth_scheme_prefix_size = 6;
+					std::string authorization{iter->value().substr(alt_method + basic_auth_scheme_prefix_size)};
+					boost::trim(authorization);
+					log::debug("{}: Authorization value (trimmed): [{}]", fn, authorization);
+
+					constexpr auto max_creds_size = 128;
+					unsigned long size = max_creds_size;
```

Comment:
Prefer a more explicit type.

Reply:
Think you had changed this, but was lost in rebase. Will change to what you had.